### PR TITLE
PlayList-Page

### PR DESCRIPTION
**Pull request by SaurabhChawda**
Added Playlist in Video Library.
List of Features.
1. User can create Playlist and add Video.
2. User can delete Playlist.
3. User can remove video from Playlist.
Screenshot (373)
Screenshot (374)

### CODE REVIEW

File: src/components/PlayListModal/playListModal.css [hunk 1]
```diff
@@ -19,35 +19,48 @@
...
   padding: 0.5rem;
   font-size: var(--font-size-small);
 }
-.modal__btn:hover {
+.playList-modal__btn:hover {
   background-color: var(--primary-color);
 }
+
+.playList-item--container {
+  display: flex;
```
Comment (paarthbajaj) [inline src/components/PlayListModal/playListModal.css:56]: There is a lot of code repetition in CSS. You can remove some code like display: flex and create a utility class for it.

File: src/Contexts/UserContext.jsx [hunk 2]
```diff
@@ -22,11 +23,12 @@ const UserProvider = ({ children }) => {
           const {
             data: { watchlater },
           } = await axios.post("/api/user/watchlater", { video }, { headers: { authorization: authState.token } });
-          console.log(watchlater);
         } catch (err) {
-          console.log(err);
+          toast.error("Failed to Add");
```
Comment (paarthbajaj) [inline src/Contexts/UserContext.jsx:27]: This is good. I only console log the errors but doing this is a great practice. Well done.

File: src/Contexts/UserContext.jsx [hunk 3]
```diff
@@ -95,13 +95,102 @@ const UserProvider = ({ children }) => {
...
+          );
+          dispatch({ type: "Create_PlayList", payload: playlists });
+        } catch (error) {
+          console.log(error);
+        }
+      })();
+    }
+  };
+  const DeletePlaylist = (playList) => {
+    if (authState.isUserLoggedIn) {
```
Comment (paarthbajaj) [inline src/Contexts/UserContext.jsx:139]: Really good, that you have implemented this in the early phase of development.

**Review by paarthbajaj — APPROVED** [review]
Code is well written and sanity checks that you have applied are very great.Make sure you don't write too much repetitive CSS and don't forget to removed the commented code before submission. Well done!

File: src/Contexts/UserContext.jsx [hunk 4]
```diff
@@ -95,13 +95,102 @@ const UserProvider = ({ children }) => {
...
+      dispatch({ type: "Delete_PlayList", payload: playList });
+      (async () => {
+        try {
+          const {
+            data: { playlists },
+          } = await axios.delete(`/api/user/playlists/${playList._id}`, {
```
Comment (parthgabani777) [inline src/Contexts/UserContext.jsx:145]: You can use separate(services) file for calling API

File: src/Pages/PlayList-Page/Components/Header.css [hunk 5]
```diff
@@ -0,0 +1,19 @@
+.header--container {
```
Comment (parthgabani777) [inline src/Pages/PlayList-Page/Components/Header.css:1]: You don't have to import header.css on every component you can directly import once on App/Route Components.

**Review by parthgabani777 — APPROVED** [review]
Code looks working and consistent.
Good use of toast message

File: src/components/PlayListModal/playListModal.css (hunk 1 above)
Comment (SaurabhChawda) [reply src/components/PlayListModal/playListModal.css:56]: Sure,I Will update.

**Comment by SaurabhChawda** [comment]
> Code is well written and sanity checks that you have applied are very great.Make sure you don't write too much repetitive CSS and don't forget to removed the commented code before submission. Well done!

Thankyou so much for your deep review, I Will update code as per your Suggestion.

File: src/Contexts/UserContext.jsx (hunk 4 above)
Comment (SaurabhChawda) [reply src/Contexts/UserContext.jsx:145]: OK, I will create.

File: src/Pages/PlayList-Page/Components/Header.css (hunk 5 above)
Comment (SaurabhChawda) [reply src/Pages/PlayList-Page/Components/Header.css:1]: Okay,ThankYou for Review.

**Pull request merged**